working/0323-null-aware-elements/feature-specification.md

```diff
@@ -466,24 +499,24 @@ add two new cases to that procedure:
 
 [unified-dynamic-element]: https://github.com/dart-lang/language/blob/main/accepted/2.3/unified-collections/feature-specification.md#to-evaluate-a-collection-element
 
-*   If `element` is a null-aware `expressionElement` with expression `e`:
+*   If `element` is a `nullAwareExpressionElement` with expression `e`:
 
     *   Evaluate `e` to `v`.
 
     *   If `v` is not `null` then append it to *result*. *Else the `null` is
         discarded.*
 
-*   Else, if `element` is a null-aware `mapElement` with entry `k: v`:
+*   Else, if `element` is a `nullAwareMapElement` with entry `k: v`:
 
     *   Evaluate `k` to a value `kv`.
 
-    *   If `element` is a key null-aware element and `kv` is `null`, then stop.
-        Else continue...
+    *   If `element` has a null-aware key and `kv` is `null`, then stop. Else
+        continue...
 
     *   Evaluate `v` to a value `vv`.
 
-    *   If `element` is a value null-aware element and `vv` is `null`, then
-        stop. Else continue...
+    *   If `element` has a null-aware value and `vv` is `null`, then stop. Else
+        continue...
```

Comment:
Here we probably want to coerce both after the `null` check, although it's probably usually the value where it makes sense (functions are bad keys).